### PR TITLE
[stdlib] Remove FileCheck for `test_bfloat16.mojo`

### PR DESCRIPTION
**Pull request by gabrieldemarmiesse**
Related to #2024

### CODE REVIEW

File: stdlib/test/builtin/test_bfloat16.mojo [hunk 1]
```diff
@@ -35,46 +34,58 @@ def test_methods():
     assert_almost_equal(BFloat16(2.0), 2.0)
 
 
-fn test_bf_primitives():
+def test_bf_primitives():
```
Comment (Moosems) [inline stdlib/test/builtin/test_bfloat16.mojo:37]: Why the change to using `def` instead of `fn`?

File: stdlib/test/builtin/test_bfloat16.mojo [hunk 2]
```diff
@@ -35,46 +34,58 @@ def test_methods():
...
+    assert_equal(a == b, a_hp == b_hp)
+    assert_equal(a != b, a_hp != b_hp)
+    assert_equal(a <= b, a_hp <= b_hp)
+    assert_equal(a >= b, a_hp >= b_hp)
 
-def main():
-    # CHECK: 33.0
-    print(
+
+def check_float64_values():
```
Comment (Moosems) [inline stdlib/test/builtin/test_bfloat16.mojo:56]: Done here as well

**Review by Moosems — COMMENTED** [review]
Quick question

File: stdlib/test/builtin/test_bfloat16.mojo (hunk 2 above)
Comment (gabrieldemarmiesse) [reply stdlib/test/builtin/test_bfloat16.mojo:56]: Using `assert_*` function means that the function can now raise. So we have two choices, either we use `raising` to the function, either we use `def` instead of `fn`. I chose def, but I don't have a strong opinion about this

File: stdlib/test/builtin/test_bfloat16.mojo [hunk 3]
```diff
@@ -10,11 +10,10 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 # ===----------------------------------------------------------------------=== #
-# RUN: %mojo %s | FileCheck %s
+# RUN: %mojo -debug-level full %s
```
Comment (JoeLoser) [inline stdlib/test/builtin/test_bfloat16.mojo:13]: Similar comment about dropping the `-debug-level full` as in other PRs.

File: stdlib/test/builtin/test_bfloat16.mojo [hunk 4]
```diff
@@ -35,46 +34,58 @@ def test_methods():
...
-    print(a * b)
-    print(a == b)
-    print(a != b)
-    print(a <= b)
-    print(a >= b)
+    # higher precision
+    var a_hp = a.cast[DType.float64]()
+    var b_hp = b.cast[DType.float64]()
 
+    assert_almost_equal(a + b, (a_hp + b_hp).cast[DType.bfloat16]())
```
Comment (JoeLoser) [inline stdlib/test/builtin/test_bfloat16.mojo:46]: Thanks for adding these!  Surprised the original code just had naked `print`s without any `CHECK` directives 🤷

File: stdlib/test/builtin/test_bfloat16.mojo [hunk 5]
```diff
@@ -35,46 +34,58 @@ def test_methods():
...
+                        _type = __mlir_type[`!pop.scalar<bf16>`],
+                        value = __mlir_attr[
+                            `#pop.simd<"nan"> : !pop.scalar<bf16>`
+                        ],
+                    ]()
+                )
             )
-        )
+        ),
+        "nan",
```
Comment (JoeLoser) [inline stdlib/test/builtin/test_bfloat16.mojo:82]: 👍 good to compare the string `nan` here since can't use the `math` module in the open source part of the standard library right now.

**Review by JoeLoser — APPROVED** [review]
Looks good, thank you so much for pushing on all of this `FileCheck` removal!  Happy to merge after we remove the `-debug-level full` part in the `RUN` directive.

File: stdlib/test/builtin/test_bfloat16.mojo (hunk 3 above)
Comment (gabrieldemarmiesse) [inline stdlib/test/builtin/test_bfloat16.mojo:13]:
```suggestion
# RUN: %mojo-no-debug %s
```
